master/internal/stream/system.go

```diff
@@ -524,37 +547,43 @@ func startup[T stream.Msg, S any](
 	state.Subscription.Configure(filter)
 
 	// Scan for historical msgs matching newly-added subscriptions.
-	var newmsgs []*websocket.PreparedMessage
+	var newmsgs []interface{}
 	newmsgs, err = state.CollectStartupMsgs(ctx, user, known, *spec)
 	if err != nil {
 		return nil, err
 	}
 	msgs = append(msgs, newmsgs...)
-	return msgs, nil
+
+	preparedMsgs := make([]interface{}, 0, len(msgs))
+	for _, msg := range msgs {
+		preparedMsgs = append(preparedMsgs, prepare(msg))
+	}
+
```

Comment:
Is this copy of `msgs` necessary?  It looks like maybe it was in an earlier version of the code, but it isn't anymore?

Reply:
still necessary, though I'm realizing I can amortize this stuff inside of the trial CollectStartupMsgs instead of here